Xamarin.Forms.Platform.UAP/SwitchRenderer.cs

```diff
@@ -53,5 +69,54 @@ void UpdateFlowDirection()
 		{
 			Control.UpdateFlowDirection(Element);
 		}
+
+		void UpdateOnColor()
+		{
+			if (Control == null)
+				return;
+
+			var grid = Control.GetChildren<Windows.UI.Xaml.Controls.Grid>().FirstOrDefault();
+			var groups = Windows.UI.Xaml.VisualStateManager.GetVisualStateGroups(grid);
+			foreach (var group in groups)
+			{
+				if (group.Name != "CommonStates")
+					continue;
+
+				foreach (var state in group.States)
+				{
+					if (state.Name != "PointerOver")
+						continue;
+
+					foreach (var timeline in state.Storyboard.Children.OfType<ObjectAnimationUsingKeyFrames>())
+					{
+						var property = Storyboard.GetTargetProperty(timeline);
+						var target = Storyboard.GetTargetName(timeline);
+						if ((target == "SwitchKnobBounds") && (property == "Fill"))
+						{
+							var frame = timeline.KeyFrames.First();
+
+							if (_originalOnColor == null)
+								_originalOnColor = frame.Value;
+
+							if (Element.IsSet(Switch.OnColorProperty) && !Element.OnColor.IsDefault)
+								frame.Value = new SolidColorBrush(Element.OnColor.ToWindowsColor()) { Opacity = .7 };
```

Comment:
why 70% opacity?

Reply:
I have taken this (with permission) from the Xamarin Community Toolkit (https://github.com/xamarin/XamarinCommunityToolkit/blob/master/Toolkit/Effects/Switch/SwitchChangeColor.uwp.cs).

This is probably because this is the default. @Depechie ?

I have another PR with the selected or unselected tab colors (https://github.com/xamarin/Xamarin.Forms/pull/4899/files#diff-a3c5e562833b067b956c34f89bd0ebbbR581), maybe implement it more like that? Capturing the default unselected color first?

Reply:
Good question, actually not sure what the default look & feel is for UWP.
If you are indeed able to get the setting from somewhere, I guess that will be better than using a fixed value.